llama_index/llms/huggingface.py

```diff
@@ -294,3 +306,160 @@ def stream_chat(
         prompt = self._messages_to_prompt(messages)
         completion_response = self.stream_complete(prompt, formatted=True, **kwargs)
         return stream_completion_response_to_chat_response(completion_response)
+
+
+def chat_messages_to_conversational_kwargs(
+    messages: Sequence[ChatMessage],
+) -> Dict[str, Any]:
+    """Convert ChatMessages to keyword arguments for Inference API conversational."""
+    if len(messages) % 2 != 1:
+        raise NotImplementedError("Messages passed in must be of odd length.")
+    last_message = messages[-1]
+    kwargs: Dict[str, Any] = {
+        "text": last_message.content,
+        **last_message.additional_kwargs,
+    }
+    if len(messages) != 1:
+        kwargs["past_user_inputs"] = []
+        kwargs["generated_responses"] = []
+        for user_msg, assistant_msg in zip(messages[::2], messages[1::2]):
```

Comment:
Is it a safe assumption to assume `user_msg` and `assistant_msg` will be alternating in the `messages` like this?

Reply:
Likely not a safe assumption. I would check message.role

Reply:
Although the llama utils make that assumption lol so maybe it's fine

Reply:
Okay, thanks for looking into this! 👌 

Fwiw, a few lines down I do validate an odd length and the `role` with a `NotImplementedError`, so we check the assumption holds